charmcraft/linters.py

```diff
@@ -325,10 +347,42 @@ def run(self, basedir: pathlib.Path) -> str:
 
         try:
             with filepath.open("rt", encoding="utf8") as fh:
-                yaml.safe_load(fh)
+                content = dict(yaml.safe_load(fh))
         except Exception:
             return self.Result.ERROR
 
+        actions_names = list(content.keys())
+
+        result_actions, text_actions = check_naming_convention(actions_names, "actions")
```

Comment:
I think I'd rather have a separate naming conventions linter here that covers both the actions and the config items. Then you can use the `url` field for the link to the style guide, and it can be used to cover other things (e.g. charm names).